rules/memory-leak/memory-leak.test.js

```diff
@@ -0,0 +1,22 @@
+const { RuleTester } = require('eslint');
+const rule = require('./memory-leak');
+
+const ruleTester = new RuleTester();
+
+const expectedError = { ruleId: 'memory-leak', message: 'Event listener is never removed' };
+ruleTester.run('memory-leak', rule, {
+  valid: [
+    'console.log("this is fine")',
+    'object.on("$destroy", console.log)',
+    'response.on("data", console.log)',
+    'stdout.on("close", console.log)',
+    'foo.on("some-event", console.log); foo.off("some-event")',
+    '$scope.on("some-event", console.log); $scope.off()'
```

Comment:
Can we add here all cases which were already working from description of
https://github.com/sirensolutions/eslint-plugin-siren/issues/2